spartan_parallel/src/sumcheck.rs

```diff
@@ -72,89 +71,43 @@ impl SumcheckInstanceProof {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct ZKSumcheckInstanceProof {
-  comm_polys: Vec<CompressedGroup>,
-  comm_evals: Vec<CompressedGroup>,
-  proofs: Vec<DotProductProof>,
+pub struct ZKSumcheckInstanceProof<S: SpartanExtensionField> {
+  proofs: Vec<DotProductProof<S>>,
 }
```

Comment:
In our usage of spartan, we don't care about zero knowledge property. Moreover, it's also difficult to build zero knowledge sumcheck protocol based on hash function. Therefore, we can get rid of (i.e. remove) types like `ZKSumcheckInstanceProof` and only keep `SumcheckInstanceProof`.